### DISCUSSION TITLE
XMLSec 1.3.6

### DISCUSSION THREAD

**Discussion by lsh123** [lsh123/xmlsec#845] [Announcements]
The XML Security Library 1.3.6 release includes the following changes:

- (xmlsec-openssl) Fixed build if OpenSSL 3.0 doesn't have engines support enabled.
- (xmlsec-mscng, xmlsec-mscrypto) Added support for multiple trusted certs with the same subject.
- (windows) Disabled iconv support by default (use 'iconv=yes' option for 'configure.js' to re-enable it).
- Several other small fixes (see [more details](https://github.com/lsh123/xmlsec/commits/master)).

Thanks for bug reports!

Aleksey

<hr /><em>This discussion was created from the release <a href='https://github.com/lsh123/xmlsec/releases/tag/1.3.6'>XMLSec 1.3.6</a>.</em>

**Comment by JoshuaNitschke** [comment]
Hi,

I apologize in advance if I am being dense and I don't feel like I know what I am doing enough to rule out this being a dumb mistake on my end.

I'm trying to upgrade an existing Dockerfile where I build xmlsec and its dependencies from source on Rocky 8 to the newest version of xmlsec and openssl 3.4, and the build is complaining that the `opensslv.h` doesn't exist even though it clearly does and I'm wondering if this is related to "Fixed build if OpenSSL 3.0 doesn't have engines support enabled." and wondering if I am doing something wrong, or if the issue is perhaps not fully fixed?

<img width="1528" alt="image" src="https://github.com/user-attachments/assets/d3234fe1-f405-4c88-94a1-d518a7196762">

Is this a bug?  Is it related to this at all? "(xmlsec-openssl) Fixed build if OpenSSL 3.0 doesn't have engines support enabled."

These are my CFLAGS:
    CFLAGS='-std=c99 -lpthread -fPIC' CC=gcc ./configure --enable-static --enable-static-linking --disable-shared --enable-openssl3-engines --with-openssl=/usr/local/compiled --with-libxslt=/usr/local/compiled  --with-libxml=/usr/local/compiled --with-default-crypto=openssl --disable-crypto-dl --disable-apps-crypto-dl --with-threads=off --with-thread-alloc==off --prefix=/usr/local/compiled && \
    make && make install

**Last working version pair was:**
OPENSSL_VERSION=1.1.1m
XMLSEC1_VERSION=1.2.32

I added --enable-openssl3-engines to CFLAGS just to see if it would fix this issue.

**Comment by lsh123** [comment]
In the error, you missed the part "OR": the script checks for presence of the openssl header file as well as openssl libraries. Can you check/confirm that you have those?

Checks:

https://github.com/lsh123/xmlsec/blob/master/configure.ac#L706-L731

**Reply by JoshuaNitschke** [reply]
Oh, I misunderstood the meaning of "or" here (that only one of these three would need to exist).  Thank you!

**Reply by lsh123** [reply]
Does it work for you now?

**Reply by JoshuaNitschke** [reply]
I'm sorry for the delay, but I haven't had a chance to look at this again just yet.  I will check the other two files and if this error is false, I will let you know, and if its not I will have to look at what has changed with building opensll that is causing the issue :)
